jupyterlab_hdf/util.py

```diff
@@ -64,59 +64,77 @@ def dsetChunk(dset, ixstr=None, subixstr=None, min_ndim=None):
 
 ## create dicts to be returned by the various api
 def hobjAttrsDict(hobj):
-    return dict((
-        *hobj.attrs.items(),
-    ))
+    return dict((*hobj.attrs.items(),))
+
 
 def hobjContentsDict(hobj, content=False, ixstr=None, min_ndim=None):
-    return dict((
-        # ensure that 'content' is undefined if not explicitly requested
-        *((('content', hobjMetaDict(hobj, ixstr=ixstr, min_ndim=min_ndim)),) if content else ()),
-        *_hobjDict(hobj).items(),
-        ('uri', hobj.name),
-    ))
+    return dict(
+        (
+            # ensure that 'content' is undefined if not explicitly requested
+            *((("content", hobjMetaDict(hobj, ixstr=ixstr, min_ndim=min_ndim)),) if content else ()),
+            *_hobjDict(hobj).items(),
+            ("uri", hobj.name),
+        )
+    )
+
 
 def hobjMetaDict(hobj, ixstr=None, min_ndim=None):
-    d = dict((
-        *_hobjDict(hobj).items(),
-        ('attributeCount', len(hobj.attrs))
-    ))
-
-    if d['type'] == 'dataset':
-        return dict(sorted((
-            *d.items(),
-            *_dsetMetaDict(hobj, ixstr=ixstr, min_ndim=min_ndim).items(),
-        )))
-    elif d['type'] == 'group':
-        return dict(sorted((
-            *d.items(),
-            ('childrenCount', len(hobj))
-        )))
+    d = dict((*_hobjDict(hobj).items(), ("attributes", [_attrMetaDict(hobj.attrs.get_id(k)) for k in sorted(hobj.attrs.keys())])))
+
+    if d["type"] == "dataset":
+        return dict(
+            sorted(
+                (
+                    *d.items(),
+                    *_dsetMetaDict(hobj, ixstr=ixstr, min_ndim=min_ndim).items(),
+                )
+            )
+        )
+    elif d["type"] == "group":
+        return dict(sorted((*d.items(), ("childrenCount", len(hobj)))))
     else:
         return d
 
+
 def hobjType(hobj):
     if isinstance(hobj, h5py.Dataset):
-        return 'dataset'
+        return "dataset"
     elif isinstance(hobj, h5py.Group):
-        return 'group'
+        return "group"
     else:
-        return 'other'
+        return "other"
+
+
+def _attrMetaDict(attrId):
+    return dict(
+        (
+            ("name", attrId.name),
+            ("dtype", attrId.dtype.str),
+            ("shape", attrId.shape),
```

Comment:
Every attr id is guaranteed to have a `.shape`?

Reply:
It appears so: https://api.h5py.org/h5a.html#h5py.h5a.AttrID

After several tests, I have yet to find one that has no shape.